flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisProducer.java

```diff
@@ -326,6 +366,29 @@ private void checkAndPropagateAsyncError() throws Exception {
 		}
 	}
 
+	/**
+	 * If the internal queue of the {@link KinesisProducer} gets too long,
+	 * flush some of the records until we are below the limit again.
+	 * We don't want to flush _all_ records at this point since that would
+	 * break record aggregation.
+	 */
+	private void enforceQueueLimit() {
```

Comment:
Would like to request one more slight change here:
Let this method return a boolean that indicates whether or not flushing occurred.

The caller of this method can then use the flag to decide whether or not `checkAndPropagateAsyncError` is required.

Reply:
✔